### PR TITLE
BUG: signal: Fix validation of the zi shape in sosfilt.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**

### CODE REVIEW

**Comment by Eric89GXL** [comment]
LGTM, +1 for merge

File: scipy/signal/signaltools.py [hunk 1]
```diff
@@ -2443,19 +2445,18 @@ def sosfilt(sos, x, axis=-1, zi=None):
     if m != 6:
         raise ValueError('sos array must be shape (n_sections, 6)')
 
-    if zi is not None:
-        use_zi = True
+    use_zi = zi is not None
+    if use_zi:
         zi = np.array(zi)
```
Comment (rgommers) [inline scipy/signal/signaltools.py]: Shouldn't this also be ``asarray``?

File: scipy/signal/tests/test_signaltools.py [hunk 2]
```diff
@@ -1440,6 +1440,38 @@ def test_initial_conditions(self):
...
+
+        # Apply the filter to x in two stages.
+        y1, z1 = sosfilt(sos, x[:, :5, :], axis=axis, zi=z0)
+        y2, z2 = sosfilt(sos, x[:, 5:, :], axis=axis, zi=z1)
+
+        # y should equal yf, and z2 should equal zf.
+        y = np.concatenate((y1, y2), axis=axis)
+        assert_allclose(y, yf, rtol=1e-10, atol=1e-13)
+        assert_allclose(z2, zf, rtol=1e-10, atol=1e-13)
+
```
Comment (rgommers) [inline scipy/signal/tests/test_signaltools.py]: Maybe an ``assert_raises`` for incorrect shape here for completeness?

**Comment by WarrenWeckesser** [comment]
Thank Ralf.  I made the changes you suggested.

**Comment by rgommers** [comment]
Okay, in it goes. Thanks @WarrenWeckesser, @Eric89GXL

**Pull request merged**